src/files/rsql-filter-expression.ts

```diff
@@ -111,16 +127,32 @@ export class RSQLFilterExpression {
           filterString += encodeURIComponent('<=') + valueString;
           break;
         case Operators.StartsWith:
-          filterString += '==' + encodeURIComponent(quote(`${valueString}*`));
+          filterString +=
+            '==' +
+            (buildOptions.encodeValues
+              ? encodeURIComponent(quote(`${valueString}*`))
+              : encodedQuote(`${valueString}*`));
           break;
         case Operators.EndsWith:
-          filterString += '==' + encodeURIComponent(quote(`*${valueString}`));
+          filterString +=
+            '==' +
+            (buildOptions.encodeValues
```

Comment:
Instead of repeating the logic:
IF encodeValues
quote(value)
ELSE
encodeQuote(value)

it would be better to create utility method which will do the logic and use it in all places.

Reply:
This has been done with a new `buildValueString` function.  Thank you for the suggestion!